### DISCUSSION TITLE
Error when loading drives when running extension in a K8S cluster

### DISCUSSION THREAD

**Discussion by BarryDrez** [eclipse-theia/theia#12536] [General]
Hi experts,

I am seeing many errors when trying to open files in my Theia extension:

logger-protocol.ts:113 2023-05-16T12:08:43.151Z root ERROR Error when loading drives. Error: spawn lsblk ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

![image](https://github.com/eclipse-theia/theia/assets/24534027/18bacb38-8ad9-4a99-adcf-a9bd4ac9e8b0)

The file does not open and there is no indication in the logs as to what may be the cause.

I am using Theia version 1.37.  Again, this only happens when running in a Kubernetes cluster.  When I run Theia in a Docker compose environment, everything works fine.

My Dockerfile and pod yaml are attached.

Thank you!
[Dockerfile.txt](https://github.com/eclipse-theia/theia/files/11487970/Dockerfile.txt)
[theia-edge-pod.yml.txt](https://github.com/eclipse-theia/theia/files/11487977/theia-edge-pod.yml.txt)

**Comment by BarryDrez** [comment]
@vince-fugnitto I opened this 2 days ago.  Not getting any feedback.  I may need to reopen the issue,

**Reply by vince-fugnitto** [reply]
@BarryDrez thank you for the discussion, people try to help or answer if they have some experience or an idea about the problem. Unfortunately I do not believe anyone encountered the issue before in either docker or kubernetes so you may need to debug on your side.

The error is thrown here:

https://github.com/eclipse-theia/theia/blob/2f8b8b65d46fe1299736e7b55161c79e9de5e759/packages/filesystem/src/browser/file-tree/file-tree-model.ts#L74-L82

The backend implementation of `getDrives` is the following:

https://github.com/eclipse-theia/theia/blob/2f8b8b65d46fe1299736e7b55161c79e9de5e759/packages/core/src/node/env-variables/env-variables-server.ts#L72-L83

**Reply by BarryDrez** [reply]
Thank you @vince-fugnitto

**Reply by BarryDrez** [reply]
@vince-fugnitto Can you please point me to a good set of instructions for configuring Kubernetes so that I can run my Theia extension in a pod?  There is obviously something wrong with my environment as I am getting the above errors on this line: const drives = await this.environments.getDrives(); 
I am unable to edit any files in my workspace.  Do I need an ephemeral volume for my workspace?
Thank you
